### DISCUSSION TITLE
In the tutorial for the Next.js integration at https://www.ory.sh/docs/getting-started/integrate-auth/nextjs the pages router is used, however my project uses the newer app router. The app router no longer provides the required api directory, so how do I integrate ory?

### DISCUSSION THREAD

**Discussion by GWnbsp** [ory/.github#122] [General]
In the tutorial for the Next.js integration at https://www.ory.sh/docs/getting-started/integrate-auth/nextjs the pages router is used, however my project uses the newer app router. The app router no longer provides the required api directory, so how do I integrate ory?

**Comment by vinckr** [comment]
Hello @GWnbsp 

we are working on an updated next.js example here: https://github.com/ory/elements/pull/240/files - should be available soon.

**Reply by GWnbsp** [reply]
yeah
